### ISSUE TITLE
Balance Error while Running Version 2.2.3

### ISSUE THREAD

**Issue by cormo1990** [helium/helium-ledger-cli#31]
Hello, i'm receiving the message "request error" while i'm trying to check the balance of my wallet.
Can someone kindly help me? i've checked the other replies but no one works for me.
thanks in advance.
![ledgererror](https://github.com/helium/helium-ledger-cli/assets/36590321/937ad6ae-e384-4cae-b4d1-44bc1950210c)

**Comment by Matt88888888** [comment]
Hey Cormo1990 - I have a similar but far worse situation!  I had the exact same issue as you above yesterday and then updated my Firmware and Apps on the Ledger Nano X this morning and am totally screwed.  When you update to latest Apps, the "Helium" only App disappears and you have 2 different Apps on your Ledger:  "Helium-Solana" and "Solana".  Neither of which will connect to the helium-ledger-cli application on my computer.  So I went from getting the exact error you posted above of Balance HNT request error to a new error of something like "Connected to Ledger but Helium App not running" or similar.  Unbelievable!!!!  Setup a CLI Cold storage wallet for added security and now I've most likely LOST ACCESS TO ALL HNT!  Everyone always says put your crypto in cold storage, but what they really should say is NEVER USE 3rd Party Software with Ledger!  I regret ever migrating any HNT over to CLI.  I should have left it in the Helium App for safe keeping...

**Comment by Matt88888888** [comment]
![Example](https://github.com/helium/helium-ledger-cli/assets/110575029/472e947f-c127-4261-a77d-db06ab3641a7)
This was a few moments before I upgraded the Ledger Live software and Ledger Nano X... which ultimately made it so my helium-ledger-cli prgm on PC won't run/connect anymore.  All I have is my Helium Address recorded, but don't believe when we setup the helium-ledger-cli in the first place, it provided any seed phrases or similar... I believe the Ledger handled all of that.  So by not connecting to the Ledger now, access to HNT is gone forever.

**Comment by cormo1990** [comment]
> ![Example](https://private-user-images.githubusercontent.com/110575029/290999776-472e947f-c127-4261-a77d-db06ab3641a7.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI3NDQ5MDEsIm5iZiI6MTcwMjc0NDYwMSwicGF0aCI6Ii8xMTA1NzUwMjkvMjkwOTk5Nzc2LTQ3MmU5NDdmLWMxMjctNDI2MS1hNzdkLWRiMDZhYjM2NDFhNy5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxNlQxNjM2NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YTRlMWU4NDM2NzU5YzY2NWJhOTNjNGNmZjE4NTk0M2ViNTkyYzE5YjNlODBhYjNhMjVhZjJlZWVlODVkODFkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UyShVjpsm7O68SF4azJWVswy9I8FWz4kdQU1E2OWbAE) This was a few moments before I upgraded the Ledger Live software and Ledger Nano X... which ultimately made it so my helium-ledger-cli prgm on PC won't run/connect anymore. All I have is my Helium Address recorded, but don't believe when we setup the helium-ledger-cli in the first place, it provided any seed phrases or similar... I believe the Ledger handled all of that. So by not connecting to the Ledger now, access to HNT is gone forever.

Oh so sorry for your issue! hope someone can help us.

**Comment by Matt88888888** [comment]
It doesn't look good.  It was over a year ago when I setup the CLI Wallet on the computer.  I honestly don't remember much about the setup steps.  Do you recall if we were shown a seed phrase or anything similar in the CLI Wallet setup?  I'm looking through all my records and don't have any info from the process.  What app is on your Ledger right now?  The original Helium one or the newer versions... Solana or Helium-Solana?  The only thing I'll add is I'll NEVER use a 3rd Party software with Ledger ever again for any crypto cold storage.

**Comment by cormo1990** [comment]
> It doesn't look good. It was over a year ago when I setup the CLI Wallet on the computer. I honestly don't remember much about the setup steps. Do you recall if we were shown a seed phrase or anything similar in the CLI Wallet setup? I'm looking through all my records and don't have any info from the process. What app is on your Ledger right now? The original Helium one or the newer versions... Solana or Helium-Solana? The only thing I'll add is I'll NEVER use a 3rd Party software with Ledger ever again for any crypto cold storage.

No seed phrase man...
at the moment in my ledger there are HELIUM, SOLANA and HELIUM-SOLANA.
i was reading here https://support.ledger.com/hc/en-us/articles/5924999093533-How-to-access-your-Ledger-Helium-HNT-account-via-the-Helium-HNT-Wallet-mobile-app?docs=true
that we should be able to manage HNT through Solflare but i haven't try it.

i hope that someone can reply to us.

**Comment by cormo1990** [comment]
> > It doesn't look good. It was over a year ago when I setup the CLI Wallet on the computer. I honestly don't remember much about the setup steps. Do you recall if we were shown a seed phrase or anything similar in the CLI Wallet setup? I'm looking through all my records and don't have any info from the process. What app is on your Ledger right now? The original Helium one or the newer versions... Solana or Helium-Solana? The only thing I'll add is I'll NEVER use a 3rd Party software with Ledger ever again for any crypto cold storage.
> 
> No seed phrase man... at the moment in my ledger there are HELIUM, SOLANA and HELIUM-SOLANA. i was reading here https://support.ledger.com/hc/en-us/articles/5924999093533-How-to-access-your-Ledger-Helium-HNT-account-via-the-Helium-HNT-Wallet-mobile-app?docs=true that we should be able to manage HNT through Solflare but i haven't try it.
> 
> i hope that someone can reply to us.

@Matt88888888  maybe this is the right way

<img width="1337" alt="image" src="https://github.com/helium/helium-ledger-cli/assets/36590321/59950052-eb35-4042-918a-97a182f82f74">

**Comment by Matt88888888** [comment]
Ok, I believe I've read some of that over the past 24 hrs also.  If you upgrade the firmware on your Ledger, you may very well lose the older Helium App.  The fact that you have Helium, Helium-Solana, Solana all on your Ledger means maybe you're not at the latest firmware for ledger... be careful.  The part that I'm confused about is in the "migration" instructions and videos, there's a part where you put your CLI Wallet Address into the webpage and it binds or seeds to a new Solana Address.  After that, not sure how we view our balances?  Via Solflare?  Is that an App for our IOS/phones?   Why the hell does it have to be so messy and conviluted?!

**Comment by cormo1990** [comment]
> Ok, I believe I've read some of that over the past 24 hrs also. If you upgrade the firmware on your Ledger, you may very well lose the older Helium App. The fact that you have Helium, Helium-Solana, Solana all on your Ledger means maybe you're not at the latest firmware for ledger... be careful. The part that I'm confused about is in the "migration" instructions and videos, there's a part where you put your CLI Wallet Address into the webpage and it binds or seeds to a new Solana Address. After that, not sure how we view our balances? Via Solflare? Is that an App for our IOS/phones? Why the hell does it have to be so messy and conviluted?!

Hi Mate, with the amazing help of one admin into discord i solved and finally have my HNT!

i suggest you to follow the migration like descripted here:
https://docs.helium.com/solana/migration/ledger
then log into solflare (with solana wallet of the ledger) and that's it.
you will find your HNT.

feel free to ask if you need help.

cheers

**Comment by Matt88888888** [comment]
Still nothing makes sense to me here.  This whole cold storage setup was a joke and kicking myself for "trying to be secure."  I have an HNT address that was originally created by the helium-ledger-cli prgm and needed to have the Ledger connected with the "original" Helium App (now NOT on Ledger, because it was replaced by 2 other Apps: Solana & Helium-Solana) in order to see balances, transfer in and out, etc..  No seed phrases, no keys, just a damn Ledger you connect to the helium-ledger-cli with... that should have been a warning I missed.  So now, I have an HNT address that was created by helium-ledger-cli and when I used the migration page below, it actually gave me a SOL address that my HNT address is seeded to.  I can actually click on the SOL address and see the amount of HNT associated with it but again, HAVE NO WAY TO ACCESS THE HNT Address or SOL address.  At this point, it's a total loss and going to have to chalk it up as a hard lesson.  If it's 3rd Party and seems sketch, it probably is!  The whole idea here was to have HNT secured for longterm and now it's longterm alright.  Like forever gone longterm!  LOL

**Comment by cormo1990** [comment]
> Still nothing makes sense to me here. This whole cold storage setup was a joke and kicking myself for "trying to be secure." I have an HNT address that was originally created by the helium-ledger-cli prgm and needed to have the Ledger connected with the "original" Helium App (now NOT on Ledger, because it was replaced by 2 other Apps: Solana & Helium-Solana) in order to see balances, transfer in and out, etc.. No seed phrases, no keys, just a damn Ledger you connect to the helium-ledger-cli with... that should have been a warning I missed. So now, I have an HNT address that was created by helium-ledger-cli and when I used the migration page below, it actually gave me a SOL address that my HNT address is seeded to. I can actually click on the SOL address and see the amount of HNT associated with it but again, HAVE NO WAY TO ACCESS THE HNT Address or SOL address. At this point, it's a total loss and going to have to chalk it up as a hard lesson. If it's 3rd Party and seems sketch, it probably is! The whole idea here was to have HNT secured for longterm and now it's longterm alright. Like forever gone longterm! LOL

But if you connect your ledger on solflare with SOLANA wallet, you should be able to select which wallet (HNT or SOL) add to solflare and then allow to manage it without any seed phrases or keys.

**Comment by Matt88888888** [comment]
> > Still nothing makes sense to me here. This whole cold storage setup was a joke and kicking myself for "trying to be secure." I have an HNT address that was originally created by the helium-ledger-cli prgm and needed to have the Ledger connected with the "original" Helium App (now NOT on Ledger, because it was replaced by 2 other Apps: Solana & Helium-Solana) in order to see balances, transfer in and out, etc.. No seed phrases, no keys, just a damn Ledger you connect to the helium-ledger-cli with... that should have been a warning I missed. So now, I have an HNT address that was created by helium-ledger-cli and when I used the migration page below, it actually gave me a SOL address that my HNT address is seeded to. I can actually click on the SOL address and see the amount of HNT associated with it but again, HAVE NO WAY TO ACCESS THE HNT Address or SOL address. At this point, it's a total loss and going to have to chalk it up as a hard lesson. If it's 3rd Party and seems sketch, it probably is! The whole idea here was to have HNT secured for longterm and now it's longterm alright. Like forever gone longterm! LOL
> 
> But if you connect your ledger on solflare with SOLANA wallet, you should be able to select which wallet (HNT or SOL) add to solflare and then allow to manage it without any seed phrases or keys.

Where is that?  Remember, I only have "helium-solana" and "solana" apps on my Ledger.   Here's all I see in my Solflare prgm.  Did you click on the + wallet and pick HNT?  I tried that and it says something like not enough SOL to complete.  
![Capture2](https://github.com/helium/helium-ledger-cli/assets/110575029/806418cb-a6f5-413d-b8d5-a61fb61cc3e6)

**Comment by cormo1990** [comment]
> > > Still nothing makes sense to me here. This whole cold storage setup was a joke and kicking myself for "trying to be secure." I have an HNT address that was originally created by the helium-ledger-cli prgm and needed to have the Ledger connected with the "original" Helium App (now NOT on Ledger, because it was replaced by 2 other Apps: Solana & Helium-Solana) in order to see balances, transfer in and out, etc.. No seed phrases, no keys, just a damn Ledger you connect to the helium-ledger-cli with... that should have been a warning I missed. So now, I have an HNT address that was created by helium-ledger-cli and when I used the migration page below, it actually gave me a SOL address that my HNT address is seeded to. I can actually click on the SOL address and see the amount of HNT associated with it but again, HAVE NO WAY TO ACCESS THE HNT Address or SOL address. At this point, it's a total loss and going to have to chalk it up as a hard lesson. If it's 3rd Party and seems sketch, it probably is! The whole idea here was to have HNT secured for longterm and now it's longterm alright. Like forever gone longterm! LOL
> > 
> > 
> > But if you connect your ledger on solflare with SOLANA wallet, you should be able to select which wallet (HNT or SOL) add to solflare and then allow to manage it without any seed phrases or keys.
> 
> Where is that? Remember, I only have "helium-solana" and "solana" apps on my Ledger. Here's all I see in my Solflare prgm. Did you click on the + wallet and pick HNT? I tried that and it says something like not enough SOL to complete. ![Capture2](https://private-user-images.githubusercontent.com/110575029/291012680-806418cb-a6f5-413d-b8d5-a61fb61cc3e6.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxMDgwMjcsIm5iZiI6MTcwMzEwNzcyNywicGF0aCI6Ii8xMTA1NzUwMjkvMjkxMDEyNjgwLTgwNjQxOGNiLWE2ZjUtNDEzZC1iOGQ1LWE2MWZiNjFjYzNlNi5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyMFQyMTI4NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YWM5ZGMzNGZjYzRmODQ4ZGRkNTQ2ODBlYTdhNzEwMDU4YmQyNzljZGQ2ZWQ4NGE4ZjFjMjI5ZjI5ZmRkZjI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.SSA8p_ebJ0wuSnlHMdAycF1j07Xbj6pR9F6gwd0RHPM)

Hi Sorry for the late reply...you have to choose the right wallet when you login into solflare

**Comment by Matt88888888** [comment]
Hey man, I figured it out finally… the part that was not working for me was that last step in video where you confirm the transactions or sign them?  When you going back and forth in ledger between helium-solana and solana apps.  It was sitting there and spinning at that last step.  It did that like 8 times over 2 days thinking there was something else wrong.  It finally worked yesterday and then I was able to complete that step, get into Solflare again and finally access my HNT!  I was so relieved, I immediately sold half my HNT for BTC out of frustration and will hold my other half of HNT for another year or two and see what it does.  Thanks fo replying and maybe or comments here will help someone else down the road…
